src/storage-client/src/healthcheck.rs

```diff
@@ -57,3 +69,88 @@ pub static MZ_SOURCE_STATUS_HISTORY_DESC: Lazy<RelationDesc> = Lazy::new(|| {
         .with_column("error", ScalarType::String.nullable(true))
         .with_column("details", ScalarType::Jsonb.nullable(true))
 });
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_row() {
+        let error_message = "error message";
+        let hint = "hint message";
+        let id = GlobalId::User(1);
+        let status = "dropped";
+        let row = pack_status_row(id, status, Some(error_message), 1000, Some(hint));
+
+        for (datum, column_type) in row.iter().zip(MZ_SINK_STATUS_HISTORY_DESC.iter_types()) {
+            assert!(datum.is_instance_of(column_type));
+        }
+
+        for (datum, column_type) in row.iter().zip(MZ_SOURCE_STATUS_HISTORY_DESC.iter_types()) {
+            assert!(datum.is_instance_of(column_type));
+        }
+
+        assert_eq!(row.iter().nth(1).unwrap(), Datum::String(&id.to_string()));
+        assert_eq!(row.iter().nth(2).unwrap(), Datum::String(status));
+        assert_eq!(row.iter().nth(3).unwrap(), Datum::String(error_message));
+        assert_eq!(
+            row.iter()
+                .nth(4)
+                .unwrap()
+                .unwrap_map()
+                .iter()
+                .collect::<Vec<_>>(),
```

Comment:
this is some pretty advanced rust for someone who calls themself a "noob"!

Reply:
Courtesy, rows documentation which had this very helpful snippet https://github.com/MaterializeInc/materialize/blob/main/src/repr/src/row.rs#L1232-L1234 😬 (and it's pretty similar to Scala collect!)

```
/// assert_eq!(
///     row.unpack_first().unwrap_map().iter().collect::<Vec<_>>(),
///     vec![("age", Datum::Int64(42)), ("name", Datum::String("bob"))]
```